autoload/jedi.vim

```diff
@@ -460,36 +461,180 @@ function! jedi#show_call_signatures() abort
+        call jedi#clear_call_signatures()
+    endif
+    let s:show_call_signatures_delayed_posmode = [getpos('.'), a:mode]
+    let s:show_call_signatures_timer = timer_start(
+                \ g:jedi#show_call_signatures_delay,
+                \ function('s:show_call_signatures_delayed_cb'))
+endfunction
+
 
 function! jedi#clear_call_signatures() abort
     if s:_init_python == 0
         return 1
     endif
-
+    if s:show_call_signatures_last[0] == 0
+        " No call signature displayed currently.
+        return
+    endif
     let s:show_call_signatures_last = [0, 0, '']
     PythonJedi jedi_vim.clear_call_signatures()
 endfunction
 
 
-function! jedi#configure_call_signatures() abort
-    augroup jedi_call_signatures
-    autocmd! * <buffer>
-    if g:jedi#show_call_signatures == 2  " Command line call signatures
-        autocmd InsertEnter <buffer> let g:jedi#first_col = s:save_first_col()
-    endif
-    autocmd InsertEnter <buffer> let s:show_call_signatures_last = [0, 0, '']
-    autocmd InsertLeave <buffer> call jedi#clear_call_signatures()
-    if g:jedi#show_call_signatures_delay > 0
-        autocmd InsertEnter <buffer> let b:_jedi_orig_updatetime = &updatetime
-                    \ | let &updatetime = g:jedi#show_call_signatures_delay
-        autocmd InsertLeave <buffer> if exists('b:_jedi_orig_updatetime')
-                    \ |   let &updatetime = b:_jedi_orig_updatetime
-                    \ |   unlet b:_jedi_orig_updatetime
-                    \ | endif
-        autocmd CursorHoldI <buffer> call jedi#show_call_signatures()
+let s:prev_show_call_signatures = 0
+function! jedi#configure_call_signatures(...) abort
+    if a:0
+        let old = g:jedi#show_call_signatures
+        let new = a:1
     else
-        autocmd CursorMovedI <buffer> call jedi#show_call_signatures()
+        let old = s:prev_show_call_signatures
+        let new = g:jedi#show_call_signatures
+    endif
+    if a:0 > 1
+        let g:jedi#show_call_signatures_modes = a:2
+    endif
+
+    if new == 1 && !has('conceal')
+        echohl WarningMsg
+        echom "jedi-vim's show_call_signatures=1 requires the conceal feature."
+        echohl None
+        return
+    endif
+
+    let g:jedi#show_call_signatures = new
+
+    if s:prev_show_call_signatures
+      call jedi#clear_call_signatures()
+    endif
+    if new != s:prev_show_call_signatures
+        " Cleanup.
+        if s:prev_show_call_signatures == 1
+            syn clear jediIgnore jediFatSymbol jediFat jediSpace jediFunction
+            setl conceallevel&
+        endif
+    endif
+    let s:prev_show_call_signatures = g:jedi#show_call_signatures
+
+    if g:jedi#show_call_signatures == 0
+        augroup jedi_call_signatures
+            au!
+        augroup END
+        return
+
+    elseif g:jedi#show_call_signatures == 1
+        let e = g:jedi#call_signature_escape
+        let full = e.'jedi=.\{-}'.e.'.\{-}'.e.'jedi'.e
+        let ignore = e.'jedi.\{-}'.e
```

Comment:
That was probably not like you've meant it.
Looking closer at it I don't think `ignore` should be removed here, since it is used to ignore the escape sequences after all?!  Changed in ff1ce5f for now.